### ISSUE TITLE
Dropout is changing the values it doesn't zero out

### ISSUE THREAD

**Issue by sserr23** [pytorch/pytorch#5752]
PyTorch version: 0.2.0_4

It appears that torch.nn.Dropout is multiplying the values it doesn't zero out by 1/(1 - p). (I'm assuming this isn't expected behavior?)

```
import torch
b = torch.FloatTensor([i for i in range(30)]).view(3, 2, 5)
b = torch.nn.autograd.Variable(b)
dpoint3 = torch.nn.Dropout(p=.3)
dpoint5 = torch.nn.Dropout(p=.5)
```
b is

```
Variable containing:
(0 ,.,.) = 
   0   1   2   3   4
   5   6   7   8   9

(1 ,.,.) = 
  10  11  12  13  14
  15  16  17  18  19

(2 ,.,.) = 
  20  21  22  23  24
  25  26  27  28  29
[torch.FloatTensor of size 3x2x5]
```

If I do

```dpoint3(b)```

I get

```
Variable containing:
(0 ,.,.) = 
   0.0000   1.4286   2.8571   4.2857   5.7143
   0.0000   0.0000  10.0000  11.4286  12.8571

(1 ,.,.) = 
  14.2857  15.7143  17.1429  18.5714  20.0000
   0.0000  22.8571   0.0000   0.0000  27.1429

(2 ,.,.) = 
  28.5714  30.0000   0.0000  32.8571  34.2857
   0.0000  37.1429  38.5714  40.0000  41.4286
[torch.FloatTensor of size 3x2x5]
```

If I do 

```dpoint5(b)```

I get

```
Variable containing:
(0 ,.,.) = 
   0   0   0   0   8
  10  12  14   0   0

(1 ,.,.) = 
   0  22  24   0   0
  30   0   0  36   0

(2 ,.,.) = 
  40   0  44   0   0
  50  52   0  56  58
[torch.FloatTensor of size 3x2x5]
```

**Comment by apaszke** [comment]
This is expected. This behavior is pretty standard and makes dropout a no-op during evaluation/inference (instead of scaling then).

**Comment by ksopyla** [comment]
I have the same observation and at the beginning, I was confused, thx @apaszke for confirmation.
I understand that is expected, but could you elaborate on this matter. How is this implemented or provide some useful links to articles?

**Comment by jiayouwyhit** [comment]
> This is expected. This behavior is pretty standard and makes dropout a no-op during evaluation/inference (instead of scaling then).

@apaszke I also have the same observation. But I am still a bit confused. Why is such kind of scaling of the values necessary? Could you give more clarification on this? Thanks.

**Comment by jiayouwyhit** [comment]
A useful link:
https://stackoverflow.com/questions/34597316/why-input-is-scaled-in-tf-nn-dropout-in-tensorflow
